app/src/main/java/org/oppia/android/app/translation/AppLanguageResourceHandler.kt

```diff
@@ -160,6 +161,26 @@ class AppLanguageResourceHandler @Inject constructor(
     }
   }
 
+  /**
+   * Returns a human-readable, localized representation of the specified [OppiaLanguage].
+   *
+   * Note that the returned string is not expected to be localized to the user's current locale.
+   * Instead, it will be localized for that specific language (i.e. each language will be
+   * represented within that language to make it easier to identify when choosing a language).
+   */
+  fun computeLocalizedDisplayName(oppiaLanguage: OppiaLanguage): String {
+    return when (oppiaLanguage) {
+      OppiaLanguage.HINDI -> getLocalizedDisplayName("hi")
+      OppiaLanguage.PORTUGUESE -> getLocalizedDisplayName("pt")
+      OppiaLanguage.SWAHILI -> getLocalizedDisplayName("sw")
+      OppiaLanguage.BRAZILIAN_PORTUGUESE -> getLocalizedDisplayName("pt", "BR")
+      OppiaLanguage.UNRECOGNIZED, OppiaLanguage.LANGUAGE_UNSPECIFIED,
+      OppiaLanguage.ENGLISH -> getLocalizedDisplayName("en")
+      OppiaLanguage.ARABIC -> getLocalizedDisplayName("ar")
+      OppiaLanguage.HINGLISH -> getLocalizedDisplayName("hi", "en")
```

Comment:
Since Hinglish doesn't actually exist, we'll probably need to add an untranslated string for this called "Hinglish" and retrieve that, instead.

Similarly, I think you could actually add localized names for each language type to untranslated_strings since the expectation is that these won't change. You can compute them once (e.g. using something like Kotlin playground) to get the values to put in the strings file, and then we can just retrieve them here. This also conveniently addresses part of #3791 by removing the need for ``Locale`` below.

Reply:
Hi @BenHenning please clarify what you mean by "untranslated string" in this context, also how do i add localized names for each language type to untranslated_strings.

Reply:
The idea is adding new strings to https://github.com/oppia/oppia-android/blob/develop/app/src/main/res/values/untranslated_strings.xml for each of the languages to contain their localized names, and then referencing those instead of relying on Locale.

Reply:
Updated.